src/lib.rs

```diff
@@ -1,25 +1,45 @@
-use std::io::BufRead;
+use rayon::iter::ParallelBridge;
+use rayon::prelude::*;
+use std::fs::File;
+use std::path::Path;
 
+mod chunks;
 mod counter;
 mod key_finder;
 
 pub use counter::{Counter, KeyCount};
 pub use key_finder::KeyFinder;
 
-pub fn top_few_from_stream<T: BufRead>(
-    reader: T,
+fn sum(mut l: Counter, r: Counter) -> Counter {
+    for (key, count) in r.counts().into_iter() {
+        l.add(&key, count);
+    }
+    l
+}
+
+pub fn top_few_from_stream(
+    path: &Path,
     kf: &KeyFinder,
     num: usize,
-) -> Result<Vec<KeyCount>, anyhow::Error> {
-    let mut counter = Counter::new(num);
-    let mut s = String::new();
-    for ln in reader.lines() {
-        s.clear();
-        let ln = ln?;
-        if let Ok(key) = kf.key(&ln, &mut s) {
-            counter.add(key)
-        }
-    }
+) -> anyhow::Result<Vec<KeyCount>> {
+    let file = File::open(path)?;
+    let size = file.metadata()?.len();
+
+    let total = chunks::chunks(path, 1 << 26, size)
```

Comment:
What does this `1 << 26` constant mean?

Reply:
`1<<26` is the same as `(2 as usize).pow(26)`.